File: Python/ad.py
from advertiser import *
class Ad(BaseAdvertising):
	'This class is meant to represent an ad.'
	def __init__(self, ID, TITLE, IMGURL, LINK, ADVERTISER):
		self._id = ID
		self.__title = TITLE
		self.__imgUrl = IMGURL
		self.__link = LINK
		self.__advertiser = ADVERTISER
		self._clicks = self._views = 0
	def getTitle(self):
		return self.__title
	def setTitle(self, TITLE):
		self.__title = TITLE
	def getImgUrl(self):
		return self.__imgUrl
	def setImgUrl(self, IMGURL):
		self.__imgUrl = IMGURL
	def getLink(self):
		return self.__link
	def setLink(self, LINK):
		self.__link = LINK
	def setAdvertiser(self, ADVERTISER):
		self.__advertiser = ADVERTISER
	def incClicks(self):
		self._clicks += 1
		self.__advertiser.incClicks()
	def incViews(self):
		self._views += 1
		self.__advertiser.incViews()
	def describeMe(self):
		return Ad.__doc__

File: Cpp/ad.cpp
#include "advertiser.cpp"
using namespace std;
class Ad: public BaseAdvertising {
private:
    string title, imgUrl, link;
    Advertiser * advertiser;
public:
    Ad(int _id, string _title, string _imgUrl, string _link, Advertiser &_advertiser) {
        id = _id; title = _title; imgUrl = _imgUrl; link = _link; advertiser = &_advertiser;
    }
    string getTitle() {
        return (title);
    }
    void setTitle(string _title) {
        title = _title;
    }
    string getImgUrl() {
        return (imgUrl);
    }
    void setImgUrl(string _imgUrl) {
        imgUrl = _imgUrl;
    }
    string getLink() {
        return (link);
    }
    void setLink(string _link) {
        link = _link;
    }
    void setAdvertiser(Advertiser &_advertiser) {
        advertiser = &_advertiser;
    }
    void incClicks() {
        clicks ++;
        advertiser->incClicks();
    }
    void incViews() {
        views ++;
        advertiser->incViews();
    }
    string describeMe() {
        return ("This class is meant to represent an ad.");
    }
};
